### PR TITLE
Allow multiple webhooks and general improvements

### PR DESCRIPTION
**Pull request by yayuyokitano**
Allows the webhook scrobbler to have multiple webhooks set up for multiple different applications.
Also makes some small changes to make UX consistent.
Also fixes some random type errors and such from elsewhere in the code.
This is pretty scuffed, but it needs to be pushed with next update imo, and we have an urgent hotfix pending.
I don't think this should impact functionality of existing scrobblers.

### CODE REVIEW

File: src/core/scrobbler/webhook-scrobbler.ts [hunk 1]
```diff
@@ -100,15 +94,21 @@ export default class WebhookScrobbler extends BaseScrobbler<'Webhook'> {
...
 		try {
-			response = await timeoutPromise(timeout, promise);
-			if (response.status !== 200) {
-				return ServiceCallResult.ERROR_OTHER;
+			const responses = await timeoutPromise(
+				timeout,
+				Promise.all(promises),
+			);
+			for (const response of responses) {
+				if (response.status !== 200) {
```
Comment (inverse) [inline src/core/scrobbler/webhook-scrobbler.ts:109]: Should we add a log line here to aid in debugging what requests fail or do you think the network tab is clear enough?

File: src/ui/options/components/accounts.tsx [hunk 2]
```diff
@@ -36,7 +37,18 @@ const scrobblerPropertiesMap = {
...
 	},
+};
+
+/**
+ * Properties associated with each scrobbler that are stored as an array with items add-and-removable, and the type of input to use to edit them.
+ */
+const scrobblerArrayPropertiesMap = {
 	Webhook: {
+		applicationName: {
+			type: 'text',
```
Comment (inverse) [inline src/ui/options/components/accounts.tsx:48]: Should we change the `userApiUrl.type` to `url`?

**Review by inverse — APPROVED** [review]
Seems legit when I did some minimal testing with https://putsreq.com/.

Does it take care of migration from the previous model?

File: src/core/scrobbler/webhook-scrobbler.ts (hunk 1 above)
Comment (yayuyokitano) [reply src/core/scrobbler/webhook-scrobbler.ts:106]: Agreed, added.

File: src/ui/options/components/accounts.tsx (hunk 2 above)
Comment (yayuyokitano) [reply src/ui/options/components/accounts.tsx:48]: Long term I think its very much worth considering (as mentioned I want to completely overhaul the accounts UX in the near future). In the short term I think we should just keep it consistent with the userApiUrl type of the other scrobblers.

**Comment by yayuyokitano** [comment]
Guess I'm merging this and releasing tomorrow morning.
Its urgent, but I don't want it being pushed and something breaking while I'm asleep.

**Comment by yayuyokitano** [comment]
> Does it take care of migration from the previous model?

I forgor to reply to this💀
It does not. However it is a non-issue, because webhook has not been released yet, and the rest of the data models are unchanged.
It’s actually a major reason why I want no cersion push before this is through, despite the needed hotfix.

**Pull request merged**